tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -0,0 +1,1034 @@
+			ret = DW_TAG_enumeration_type;
+			
+			/* Split the name portion into actual name and typeID */
+			str_vect tmp = split (m.prefix(), ':');
+			
+			parsedData->name = tmp[0];
+			parsedData->typeID = tmp[1];
+			parsedData->type = m[0];
+			parsedData->members = m.suffix();
+
+			parsedData->type = stripLeading(parsedData->type, '=');
+			parsedData->members = stripLeading(parsedData->members, ':');
+			parsedData->members = striptrailing(parsedData->members, ';');
+			
+		}
+	}
+	return ret;
+}
+
+static int
+parseDwarfDie(const string line)
+{
+	
+	int ret = DW_DLV_OK;
+	/* get the type of the data of the line being parsed */
+	Parsed_Declaration_Data *parsedDeclarationData = new Parsed_Declaration_Data();
+	
+	Dwarf_Half tag = parseDeclarationLine(line, parsedDeclarationData);
+	
+	/* Parse the line further based on the type of declaration */
+	if (DW_TAG_unknown != tag) {
+		Dwarf_Die newDie;
+		double typeID = extractTypeID(parsedDeclarationData->typeID);
+		
+		double ID = _fileID + typeID;
+		
+		/* Search for the die in the createdDies before creating a new one */
+		unordered_map<double, Dwarf_Die>::iterator existingEntry = createdDies.find(ID);
+		
+		if (createdDies.end() != existingEntry) {
+			newDie = existingEntry->second;
+		} else {
+			newDie = new Dwarf_Die_s();
+			newDie->_tag = tag;
+			newDie->_context = _currentCU;
+			
+			pair<double, Dwarf_Die> result;
+			result = make_pair<double, Dwarf_Die>((double)ID, (Dwarf_Die)newDie);
+			createdDies.insert(result);
+			
+			/* If the current CU die doesn't have a child, make the first new newDie the child */
+			if (NULL == _currentCU->_die->_child) {
+				_currentCU->_die->_child = newDie;
+			}
+		}
+		switch (tag) {
+			case DW_TAG_class_type:
+			break;
+			case DW_TAG_const_type:
+			break;
+			case DW_TAG_enumeration_type:
+			break;
+			case DW_TAG_typedef:
+			break;
+			case DW_TAG_structure_type:
+			case DW_TAG_union_type:
+			break;
+			default:
+				/* No-op */
+			break;
+		}
+		
+		if (NULL != _lastDie) {
+			_lastDie->_sibling = newDie;
+		}
+		_lastDie = newDie;
+	}
+	return ret;
+}
+
+static int
+parseDwarfInfo(const char *line)
+{
+	int ret = DW_DLV_OK;
+	string data = line;
+	regex startOfFileSection (START_OF_FILE_SECTION_ROW);
+	regex fileInfoRow (FILE_INFORMATION_ROW);
+	regex dataLine (DATA_ROW);
+	regex endOfFileSection (END_OF_FILE_SECTION_ROW);
+	
+	smatch m_s;
+	string validData = "";
+	string fileName = "";
+	
+	/* state machine processes each stage of the input
+	 * 0: has not reached the start of a new file yet
+	 * 1: reached the start of a new file, parse to get info about the file, populate _currentCU
+ 	 * 2: inside of a file, parse to get information to populate dies, check if end of file reached
+ 	 */
+	if (0 == _startNewFile && regex_search(data, m_s, startOfFileSection)) {
+		_startNewFile = 1;
+		
+		/* Create one compilation unit context per file */
+		Dwarf_CU_Context *newCU = new Dwarf_CU_Context();
```

Comment:
Initialize all fields of Dwarf_CU_Context either by setting them individually or using memset() or however you prefer